hoauth2/src/Network/OAuth2/Experiment/Types.hs

```diff
@@ -721,6 +732,9 @@ instance HasTokenRequest 'ClientCredentials where
   data TokenRequest 'ClientCredentials = ClientCredentialsTokenRequest
     { scope :: Set Scope
     , grantType :: GrantTypeValue
+    , clientAssertionType :: Text
+    , clientAssertion :: BS.ByteString
+    , clientAuthenticationMethod :: ClientAuthenticationMethod
```

Comment:
need to leverage AuthMethod to determine whether append assertion to request body